src/main/kotlin/water/racingcar/GameApplication.kt

```diff
@@ -0,0 +1,52 @@
+package water.racingcar
+
+import water.racingcar.domain.Car
+import water.racingcar.dto.InputDto
+import water.racingcar.ui.GameConsoleInput
+import water.racingcar.util.RandomGenerator
+
+class GameApplication(input: InputDto) {
+    private val cars = input.carNames.map { Car(it) }.toList()
+    private val randomGenerator: RandomGenerator = RandomGenerator();
+    var times = input.times // 값은 외부에 공개하지만 클래스 외부에서 값을 변경할 수 없다.
+        private set
+
+    // 주어진 횟수 동안 각 라운드(전진 혹은 멈춤)을 진행하고 완료 후 우승자를 출력한다.
+    fun playGame() {
+        while (times > 0) {
+            times--
+            playEachRound()
+        }
+        printTotalResult()
+    }
+
+    // 각 라운드를 진행하고(전진 혹은 멈춤) 전진 상태를 출력한다.
+    private fun playEachRound() {
+        for (car in cars) {
+            car.move(randomGenerator.generateRandomNumber())
+        }
+        printEachRound()
+    }
+
+    // 우승자 출력, 공동 우승이면 여러 명을 출력한다.
+    private fun printTotalResult() {
+        val maxMovedValue = cars.maxBy { it.movedValue }?.movedValue
+        val winnerNameList = cars.filter { it.movedValue == maxMovedValue }.map { it.name }.toList()
+        val winnerNamesOutput = winnerNameList.joinToString(separator = ", ")
+        println("${winnerNamesOutput}가 최종 우승했습니다.")
```

Comment:
```suggestion
        val winnerNames = cars.filter { it.movedValue == maxMovedValue }
            .map { it.name }
            .toList()
            .joinToString(separator = ", ")
        println("${winnerNames}가 최종 우승했습니다.")
```
Stream을 쓸 때는 하나의 도트(.)마다 줄바꿈을 해서 가독성을 높였는데요. 코틀린에서도 동일하게 적용하는지 궁금하더라고요ㅋㅋㅋ 이번 피드백은 저런 방식이 코틀린에서 통하는지 정말 궁금해서 남긴거라 같이 얘기 나눠보면 좋을 거 같아요ㅋㅋㅋ